crates/e2e/src/contract_results.rs

```diff
@@ -265,3 +258,73 @@ impl<E: Environment, V: scale::Decode> CallDryRunResult<E, V> {
         String::from_utf8_lossy(&self.exec_result.debug_message).into()
     }
 }
+
+/// Result of the dry run of a contract call.
+pub struct InstantiateDryRunResult<E: Environment> {
+    /// The result of the dry run, contains debug messages if there were any.
+    pub contract_result: ContractInstantiateResult<E::AccountId, E::Balance, ()>,
+}
+
+impl<E: Environment> From<ContractInstantiateResult<E::AccountId, E::Balance, ()>>
+    for InstantiateDryRunResult<E>
+{
+    fn from(
+        contract_result: ContractInstantiateResult<E::AccountId, E::Balance, ()>,
+    ) -> Self {
+        Self { contract_result }
+    }
+}
+
+impl<E: Environment> InstantiateDryRunResult<E> {
+    /// Returns true if the dry-run execution resulted in an error.
+    pub fn is_err(&self) -> bool {
+        self.contract_result.result.is_err()
+    }
+
+    /// Returns the [`InstantiateReturnValue`] resulting from the dry-run message call.
+    ///
+    /// Panics if the dry-run message call failed to execute.
+    pub fn instantiate_return_value(&self) -> &InstantiateReturnValue<E::AccountId> {
+        self.contract_result
+            .result
+            .as_ref()
+            .unwrap_or_else(|call_err| panic!("Instantiate dry-run failed: {call_err:?}"))
+    }
+
+    /// Returns the encoded return value from the constructor.
+    ///
+    /// # Panics
+    /// - if the dry-run message instantiate failed to execute.
+    /// - if message result cannot be decoded into the expected return value type.
+    pub fn constructor_result<V: scale::Decode>(&self) -> ConstructorResult<V> {
+        let data = &self.instantiate_return_value().result.data;
+        scale::Decode::decode(&mut data.as_ref()).unwrap_or_else(|env_err| {
+            panic!("Decoding dry run result to constructor return type failed: {env_err}")
+        })
+    }
+
+    /// Returns the return value as raw bytes of the message from the dry-run.
+    ///
+    /// Panics if the dry-run message call failed to execute.
+    pub fn return_data(&self) -> &[u8] {
```

Comment:
This can already be done `constructor_result` above. That said I copied this from the `CallDryRunResult`, and I don't think this is super useful for constructors, so have deleted it.